cosmos-gui/src/app.js

```diff
@@ -0,0 +1,222 @@
+ *
+ * This file is part of fiware-tidoop (FI-WARE project).
+ *
+ * fiware-tidoop is free software: you can redistribute it and/or modify it under the terms of the GNU Affero
+ * General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.
+ * fiware-tidoop is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the
+ * implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License along with fiware-tidoop. If not, see
+ * http://www.gnu.org/licenses/.
+ *
+ * For those usages not covered by the GNU Affero General Public License please contact with
+ * francisco dot romerobueno at telefonica dot com
+ */
+
+/**
+ * cosmos-gui main app
+ *
+ * Author: frb
+ */
+
+// Module dependencies
+var express = require('express');
+var boom = require('boom');
+var stylus = require('stylus');
+var nib = require('nib');
+var config = require('../conf/cosmos-gui.json');
+var mysqlDriver = require('./mysql_driver.js');
+var OAuth2 = require('./oauth2').OAuth2;
+var cmdRunner = require('./cmd_runner.js');
+
+// Express configuration
+var app = express();
+
+app.set('views', __dirname + '/../views');
+app.set('view engine', 'jade');
+app.use(express.logger());
+app.use(stylus.middleware(
+    { src: __dirname + '/../public',
+        compile: compile
+    }
+));
+app.use(express.bodyParser());
+app.use(express.cookieParser());
+app.use(express.session({secret: "skjghskdjfhbqigohqdiouk"}));
+app.configure(function () {
+    "use strict";
+    app.use(express.errorHandler({ dumpExceptions: true, showStack: true }));
+    app.use(express.static(__dirname + '/../public'));
+});
+
+function compile(str, path) {
+    return stylus(str)
+        .set('filename', path)
+        .use(nib());
+}
+
+// Global variables
+var port = config.gui.port;
+var client_id = config.oauth2.client_id;
+var client_secret = config.oauth2.client_secret;
+var idmURL = config.oauth2.idmURL;
+var response_type = config.oauth2.response_type;
+var callbackURL = config.oauth2.callbackURL;
+var hdfsQuota = config.hdfs.quota;
+
+// Creates oauth library object with the config data
+var oa = new OAuth2(client_id,
+    client_secret,
+    idmURL,
+    '/oauth2/authorize',
+    '/oauth2/token',
+    callbackURL);
+
+// Handles requests to the main page
+app.get('/', function (req, res) {
+    var access_token = req.session.access_token;
+
+    // Check if the user had a session
+    if (access_token) {
+        // Get user information given its access token
+        oa.get(idmURL + '/user/', access_token, function (error, response) {
+            if (error) {
+                res.send(boom.internal('There was some error when getting user information from the IdM', error));
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-cosmos/commit/2f054e26fc50e1f87329bd5a79dce6904dae7407 and https://github.com/telefonicaid/fiware-cosmos/commit/a1117a2f16479934b7124e733efac4362bef4623